File: herencia-dto/HerenciaDto-Java-IvanRC/src/main/java/dto/ListaPersonasDto.java
package dto;

import com.squareup.moshi.Json;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.simpleframework.xml.ElementList;
import org.simpleframework.xml.Root;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Json(name = "personas")
@Root(name = "personas")
public class ListaPersonasDto {
    @Json(name = "personas")
    @ElementList(name = "personas", inline = true)
    List<PersonaDto> personaDtos;
}

File: herencia-dto/HerenciaDto-Kotlin-IvanRC/src/main/kotlin/dto/ListaPersonasDto.kt
package dto

import com.squareup.moshi.Json
import org.simpleframework.xml.ElementList
import org.simpleframework.xml.Root

@Json(name = "personas")
@Root(name = "personas")
data class ListaPersonasDto (
    @Json(name = "personas")
    @field:ElementList(name = "personas", inline = true)
    @param:ElementList(name = "personas", inline = true)
    val personasDto: List<PersonaDto>
)